lib/msw-config.ts

```diff
@@ -25,6 +29,13 @@ type HTTPVerb =
   | 'options'
   | 'head';
 
+type BaseHandler = (
```

Comment:
I had thought about this. Really baseHandler should be defined in mirageJS proper and this implement it to ensure any changes in the handler in mirage are respected by the pretender/msw modules.

Reply:
Also it would avoid the need to create these again for the pretender repo

Reply:
Yeah that's a great point.  In fact, I think the handlers defined in mirage are incorrect:

```
    /** Handle a GET request to the given path. */
    get<Response extends AnyResponse>(
      path: string,
      handler?: RouteHandler<Registry, Response>,
      options?: HandlerOptions
    ): void;
```

That's only one format, but they can also be just options, or just a handler, or a number of other formats (unfortunately).  

I think there's still quite a bit of cleanup / deduplication to be done, but hopefully this is at least a baby step in the right direction.